src/utilities/string_functions.js

```diff
@@ -295,20 +295,23 @@ p5.prototype.nf = function(nums, left, right) {
 };
 
 function doNf(num, left, right) {
+  let isNegative = num < 0;
+  num = Math.abs(num);
   let [leftPart, rightPart] = num.toString().split('.');
 
+
   if (typeof right === 'undefined') {
     leftPart = leftPart.padStart(left, '0');
-    return rightPart ? leftPart + '.' + rightPart : leftPart;
+    let result = rightPart ? leftPart + '.' + rightPart : leftPart;
+    return isNegative ? '-' + result : result;
   } else {
-    let roundedOff = num.toFixed(right);
-    [leftPart, rightPart] = roundedOff.toString().split('.');
+    let roundedNum = Math.round(num * Math.pow(10, right));
+    roundedNum = roundedNum / Math.pow(10, right);
+    [leftPart, rightPart] = roundedNum.toString().split('.');
     leftPart = leftPart.padStart(left, '0');
-    if(typeof rightPart === 'undefined'){
-      return leftPart;
-    }else{
-      return leftPart + '.' + rightPart;
-    }
+    let result = typeof rightPart === 'undefined' ?
+      leftPart : leftPart + '.' + rightPart.padEnd(right, '0');
```

Comment:
Was there a need to write `.padEnd`?
```suggestion
      leftPart : leftPart + '.' + rightPart;
```

can we simply add this? What you think?

Reply:
So this add padding of 0 at the right decimal place if the right is more than the digit in the right. 
if we doNf with right of 2 in 123.5 it will pad 123.50.
I guess it would be great. But tell me known what you think about this. I can change if it is not required.

Reply:
Right, when we will use `num.toFixed(right)` to roundoff the numbers, I don't think we would actually need to add padding of 0 to the right.